integration/networking/resolvconf_test.go

```diff
@@ -0,0 +1,72 @@
+package networking
+
+import (
+	"os"
+	"testing"
+
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+
+	containertypes "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/integration/internal/network"
+
+	"github.com/docker/docker/testutil/daemon"
+	"gotest.tools/v3/skip"
+)
+
+// Regression test for https://github.com/moby/moby/issues/46968
+func TestResolvConfLocalhostIPv6(t *testing.T) {
+	// No "/etc/resolv.conf" on Windows.
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+	// In rootless mode, "/etc/resolv.conf" isn't the host's, so it can't be replaced by the test.
+	skip.If(t, testEnv.IsRootless())
+
+	ctx := setupTest(t)
+
+	// Replace the test host's resolv.conf with one that only contains a loopback address.
+	// TODO(robmry) - there must be a better way!
```

Comment:
Have `resolvconf.Path()` return the value of a magic environment variable if set and start a test daemon up with that env var set? Or (this might be too clever to be practical) add an env var to set a "sysroot" path which gets prepended to all hardcoded host-absolute paths which would need to be overridden in tests. I stole this idea from GCC and clang; [it's part of their strategy for supporting cross-compilation without the hassle of chroots.](https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html#index-sysroot:~:text=%2D%2Dsysroot%3Ddir,usr/lib.)

Obviously, the environment variable would be for integration-testing purposes only. We have prior art for using environment variables this way.
https://github.com/moby/moby/blob/cff4f20c44a3a7c882ed73934dec6a77246c6323/distribution/push_v2.go#L196-L199

Another possible approach would be to unshare a mount namespace, bind-mount over /etc/resolv.conf and run a test daemon in that namespace. But the environment variable override idea is easier to implement and reason about.

Reply:
Done.